nixos/modules/services/monitoring/prometheus/default.nix

```diff
@@ -50,6 +53,38 @@ let
     (optionalString (cfg.webExternalUrl != null) "-web.external-url=${cfg.webExternalUrl}")
   ];
 
+  # This becomes the main config file for Prometheus 2
+  promConfig2 = {
+    global = cfg2.globalConfig;
+    rule_files = cfg2.ruleFiles ++ [
+      (pkgs.writeText "prometheus.rules" (concatStringsSep "\n" cfg2.rules))
+    ];
+    scrape_configs = cfg2.scrapeConfigs;
+    alerting = optionalAttrs (cfg2.alertmanagerURL != []) {
+      alertmanagers = [{
+        static_configs = [{
+          targets = cfg2.alertmanagerURL;
+        }];
+      }];
+    };
+  };
+
+  generatedPrometheus2Yml = writePrettyJSON "prometheus.yml" promConfig2;
+
+  prometheus2Yml =
+    if cfg2.configText != null then
+      pkgs.writeText "prometheus.yml" cfg2.configText
+    else generatedPrometheusYml;
+
+  cmdlineArgs2 = cfg2.extraFlags ++ [
+    "--storage.tsdb.path=${cfg2.dataDir}/data/"
+    "--config.file=${prometheusYml}"
```

Comment:
```suggestion
    "--config.file=${prometheus2Yml}"
```

Reply:
Did some test of your new module - seems to be a error here - new prometheus2 configuration will not be generated. Thanks for your work! BR Erik

Reply:
This is was integrated